sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/MetricExporterConfiguration.java

```diff
@@ -77,29 +79,58 @@ private static void configureLoggingMetrics(
 
   // Visible for testing
   @Nullable
-  static OtlpGrpcMetricExporter configureOtlpMetrics(
+  static MetricExporter configureOtlpMetrics(
       ConfigProperties config, SdkMeterProvider meterProvider) {
-    try {
-      ClasspathUtil.checkClassExists(
-          "io.opentelemetry.exporter.otlp.metrics.OtlpGrpcMetricExporter",
-          "OTLP Metrics Exporter",
-          "opentelemetry-exporter-otlp-metrics");
-    } catch (ConfigurationException e) {
-      // Squash this for now, until metrics are stable and included in the `exporter-otlp` artifact
-      // by default,
-      return null;
-    }
-    OtlpGrpcMetricExporterBuilder builder = OtlpGrpcMetricExporter.builder();
+    String protocol = OtlpConfigUtil.getOtlpProtocol(DATA_TYPE_METRICS, config);
+
+    MetricExporter exporter;
+    if (protocol.equals("http/protobuf")) {
+      try {
+        ClasspathUtil.checkClassExists(
+            "io.opentelemetry.exporter.otlp.http.metric.OtlpHttpMetricExporter",
+            "OTLP HTTP Metrics Exporter",
+            "opentelemetry-exporter-otlp-http-metrics");
+      } catch (ConfigurationException e) {
+        // Squash this for now, until metrics are stable
+        return null;
+      }
+      OtlpHttpMetricExporterBuilder builder = OtlpHttpMetricExporter.builder();
 
-    OtlpConfigUtil.configureOtlpExporterBuilder(
-        DATA_TYPE_METRICS,
-        config,
-        builder::setEndpoint,
-        builder::addHeader,
-        builder::setTimeout,
-        builder::setTrustedCertificates);
+      OtlpConfigUtil.configureOtlpExporterBuilder(
```

Comment:
The `OtlpConfigUtil` calls here and in `SpanExporterConfiguration` make me wish for a Builder interface that would contain required 4 methods and allow us to remove the duplication. But this is out of scope of this change, so entirely up to you.

Reply:
I actually went down that road first and added the builder interface to `:exporters:otlp:common`. The problem was that I was stuck getting `NoClassDefFoundError` at runtime when the `OtlpExporterBuilder` interface is referenced. The pattern in the autoconfigure module of using `ClasspathUtil.checkClassExists(...)` seems to be deterministic at runtime in that java doesn't try to load the class until after the `ClasspathUtil.checkClassExists` call. But for some reason I couldn't figure out, that behavior wasn't the same with `OtlpExporterBuilder`.